server/src/main/java/com/cloud/api/ApiServer.java

```diff
@@ -464,7 +464,11 @@ public void handle(final HttpRequest request, final HttpResponse response, final
                         responseType = param.getValue();
                         continue;
                     }
-                    parameterMap.put(param.getName(), new String[]{param.getValue()});
+                    if(parameterMap.putIfAbsent(param.getName(), new String[]{param.getValue()}) != null) {
+                        String message = String.format("Query parameter '%s' has multiple values [%s, %s]. Only the last value will be respected." +
+                            "It is advised to pass only a single parameter", param.getName(), param.getValue(), parameterMap.get(param.getName()));
+                        s_logger.warn(message);
```

Comment:
is there a joint `ApiServer`/`ApiServlet` Utility to put this warning message in, instead of copying the exact same text?
Alternatively can we change the text to indicate where/why we object to the duplicate parameter?